plane2/src/types.rs

```diff
@@ -53,23 +53,36 @@ impl BackendKeyId {
 }
 
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq, Hash, Eq)]
-pub struct ClusterId(String);
+pub struct ClusterName(String);
 
-impl ClusterId {
+impl ClusterName {
     pub fn is_https(&self) -> bool {
-        self.0 != "localhost" && !self.0.starts_with("localhost:")
+        let port = self.0.split_once(':').map(|x| x.1);
+        port.is_none() || port == Some("443")
     }
 }
 
-impl Display for ClusterId {
+impl Display for ClusterName {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         write!(f, "{}", &self.0)
     }
 }
 
-impl From<String> for ClusterId {
-    fn from(s: String) -> Self {
-        Self(s)
+impl FromStr for ClusterName {
+    type Err = &'static str;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.trim();
```

Comment:
I am tempted to say that we should not trim here, because whitespace ending up in here seems like it would be a symptom of other problems. Thoughts?

Reply:
Yeah, I think you're right. Updated!